lib/trento/application/integration/discovery/payloads/cluster/crmmon_discovery_payload.ex

```diff
@@ -0,0 +1,274 @@
+      |> validate_required_fields([:name, :fail_count, :migration_threshold])
+    end
+  end
+
+  defmodule CrmmonResource do
+    @moduledoc """
+    CrmmonResource field payload
+    """
+
+    @required_fields [
+      :id,
+      :role,
+      :agent,
+      :active,
+      :failed,
+      :blocked,
+      :managed,
+      :orphaned,
+      :failure_ignored,
+      :nodes_running_on,
+      :node
+    ]
+    use Trento.Type
+
+    deftype do
+      field :id, :string
+      field :role, :string
+      field :agent, :string
+      field :active, :boolean
+      field :failed, :boolean
+      field :blocked, :boolean
+      field :managed, :boolean
+      field :orphaned, :boolean
+      field :failure_ignored, :boolean
+      field :nodes_running_on, :integer
+
+      embeds_one :node, ResourceNode, primary_key: false do
+        field :id, :string
+        field :name, :string
+        field :cached, :boolean
+      end
+    end
+
+    def changeset(crmmon_resource, attrs) do
+      crmmon_resource
+      |> cast(attrs, [
+        :id,
+        :role,
+        :agent,
+        :active,
+        :failed,
+        :blocked,
+        :managed,
+        :orphaned,
+        :failure_ignored,
+        :nodes_running_on
+      ])
+      |> cast_embed(:node, with: &resource_node_changeset/2)
+      |> validate_required_fields(@required_fields)
+    end
+
+    defp resource_node_changeset(resource_node, attrs) do
+      resource_node
+      |> cast(attrs, [:id, :name, :cached])
+      |> validate_required_fields([])
+    end
+  end
+
+  defmodule Summary do
+    @moduledoc """
+    Summary field payload
+    """
+
+    @required_fields [:nodes, :resources, :last_change]
+    use Trento.Type
+
+    deftype do
+      embeds_one :nodes, NodesSummary do
+        field :number, :integer
+      end
+
+      embeds_one :resources, ResourceSummary do
+        field :number, :integer
+        field :blocked, :integer
+        field :disabled, :integer
+      end
+
+      embeds_one :last_change, LastChangeSummary do
+        field :time, :string
+      end
+    end
+
+    def changeset(summary, attrs) do
+      summary
+      |> cast(attrs, [])
+      |> cast_embed(:nodes, with: &nodes_changeset/2)
+      |> cast_embed(:resources, with: &resources_changeset/2)
+      |> cast_embed(:last_change, with: &last_change_changeset/2)
+      |> validate_required_fields(@required_fields)
+    end
+
+    def nodes_changeset(nodes, attrs) do
+      nodes
+      |> cast(attrs, [:number])
+      |> validate_required_fields([:number])
+    end
+
+    def resources_changeset(resources, attrs) do
+      resources
+      |> cast(attrs, [:number, :blocked, :disabled])
+      |> validate_required_fields([:number, :blocked, :disabled])
+    end
+
+    def last_change_changeset(last_change, attrs) do
+      last_change
+      |> cast(attrs, [:time])
+      |> validate_required_fields([:time])
+    end
+  end
+
+  @required_fields [
```

Comment:
Since the `deftype` needs some modules from this same file, we can only add it to the after these modules have been defined. The `@required_fields` could indeed go to the top of the file but I'd rather keep it closer to the `deftype` as it will be easier to read if they are close to each other.